src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
Ah, then would it make sense to unlock all the arg registers after the call to `genExitCode`? I *think* that it is OK to unlock more registers that one has locked, but it's been ages since I've done anything with the register lock/unlock code.